iot/api-client/end_to_end_example/cloudiot_pubsub_example_server.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2017 Google Inc. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Sample server that pushes configuration to Google Cloud IoT devices.
+
+This example represents a server that consumes telemetry data from multiple
+Cloud IoT devices. The devices report telemetry data, which the server consumes
+from a Cloud Pub/Sub topic. The server then decides whether to turn on or off
+individual devices fans.
+
+This example requires the Google Cloud Pub/Sub client library. Install it with
+
+  $ pip install --upgrade google-cloud-pubsub
+
+If you are running this example from a Compute Engine VM, you will have to
+enable the Cloud Pub/Sub API for your project, which you can do from the Cloud
+Console. Create a pubsub topic, for example
+projects/my-project-id/topics/my-topic-name, and a subscription, for example
+projects/my-project-id/subscriptions/my-topic-subscription.
+
+You can then run the example with
+
+  $ python cloudiot_pubsub_example_server.py \
+    --project_id=my-project-id \
+    --pubsub_subscription=my-topic-subscription \
+    --api_key=YOUR_API_KEY
+"""
+
+import argparse
+import base64
+import json
+import sys
+from threading import Lock
+import time
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+from oauth2client.service_account import ServiceAccountCredentials
+from google.cloud import pubsub_v1
+
+
+API_SCOPES = ['https://www.googleapis.com/auth/cloud-platform']
+API_VERSION = 'v1beta1'
+DISCOVERY_API = 'https://cloudiot.googleapis.com/$discovery/rest'
+SERVICE_NAME = 'cloudiot'
+
+
+def discovery_url(api_key):
+  """Construct the discovery url for the given api key."""
+  return '{}?version={}&key={}'.format(DISCOVERY_API, API_VERSION, api_key)
```

Comment:
API key is no longer necessary because the API is public

Reply:
Removed.